### PR TITLE
Simplify power supply naming logic

### PR DESCRIPTION
**Pull request by timmmoore**
### Requirements

* Filling out this template is required. Pull Requests without a clear description may be closed at the maintainers' discretion.

### Description
Moving power supply names  and #if POWER_SUPPLY logic from menu_info.cpp to conditionals_post.h.
Also gives the option to override the default name and to specify a different power supply name.

### Benefits

Simplifies the menu board info code in menu_info.cpp. Moves the #if logic for #define power_supply into conditionals_post.h with the rest of the #if power_supply logic.
Also gives an option to override the default power supply name, if required, i.e. power circuits that act as ATX or XBox but are not.

### CODE REVIEW

**Comment by thinkyhead** [comment]
The names are just meant to represent the PSU behavior for the sake of debug output, and not to give an accounting of the actual PSU model.

**Comment by AnHardt** [comment]
Imho POWER_SUPPLY is an unusual config option. 
Should be splittend to
POWER_SUPPLY_CONTROL or POWER_SUPPLY_CONTROL_FEATURE to indicate if a POWER_SUPPLY_PIN is to be used. And
POWER_SUPPLY_PIN_INVERTED to indicate if a HIGH pin means on or off.
// 1:'ATX', 2:'X-Box 360' is meaning exactly that but is named too special, confusing. (Help! I don't have a X-Box power supply, but with ATX on is off and off is on. Why the hell there is no hp-xyz-laptop-power-supply?)
  

@thinkyhead 
If we can previously agree about the names i'll prepare a PR.

**Comment by thinkyhead** [comment]
Everything you described sounds great. Make it so!

**Comment by timmmoore** [comment]
Take a look at this PR now, I modified it to AnHardts comments.

**Comment by thinkyhead** [comment]
Looks good! This will work.

**Comment by AnHardt** [comment]
Prepared a similar PR #14500 .
Mainly a bit different texts and dropped power supply information from menu_info_board().

**Comment by thinkyhead** [comment]
I'm melding and stitching bits of both together into a stew…. Should pass Travis one of these moments.

File: Marlin/src/Marlin.h [hunk 1]
```diff
@@ -349,8 +349,8 @@ extern millis_t max_inactive_time, stepper_inactive_time;
 
 #if HAS_POWER_SWITCH
   extern bool powersupply_on;
-  #define PSU_PIN_ON()  do{ OUT_WRITE(PS_ON_PIN, PS_ON_AWAKE); powersupply_on = true; }while(0)
-  #define PSU_PIN_OFF() do{ OUT_WRITE(PS_ON_PIN, PS_ON_ASLEEP); powersupply_on = false; }while(0)
+  #define PSU_PIN_ON()  do{ OUT_WRITE(PS_ON_PIN,  PSU_ACTIVE_HIGH); powersupply_on = true; }while(0)
+  #define PSU_PIN_OFF() do{ OUT_WRITE(PS_ON_PIN, !PSU_ACTIVE_HIGH); powersupply_on = false; }while(0)
```
Comment (AnHardt) [inline Marlin/src/Marlin.h:353]: Wow! Great.

File: Marlin/src/lcd/menu/menu_info.cpp [hunk 2]
```diff
@@ -176,13 +176,7 @@ void menu_info_board() {
...
   STATIC_ITEM(MSG_INFO_BAUDRATE ": " STRINGIFY(BAUDRATE), true); // Baud: 250000
   STATIC_ITEM(MSG_INFO_PROTOCOL ": " PROTOCOL_VERSION, true);    // Protocol: 1.0
-  #if POWER_SUPPLY == 0
-    STATIC_ITEM(MSG_INFO_PSU ": Generic", true);
-  #elif POWER_SUPPLY == 1
-    STATIC_ITEM(MSG_INFO_PSU ": ATX", true);  // Power Supply: ATX
-  #elif POWER_SUPPLY == 2
-    STATIC_ITEM(MSG_INFO_PSU ": XBox", true); // Power Supply: XBox
-  #endif
+  STATIC_ITEM(MSG_INFO_PSU ": " PSU_NAME, true);
```
Comment (AnHardt) [inline Marlin/src/lcd/menu/menu_info.cpp:179]: Honestly?
For what is this information god for? If i wanted to know i 'd have a look on the PSU, not crawling thru a menu to have the info on screen.
Belongs the PSU to the "board"?
Shouldn't `MSG_INFO_PSU` then at least read as "PSU-style".
Didn't we do this PR to get rid of the specific names.

File: Marlin/src/inc/SanityCheck.h [hunk 3]
```diff
@@ -376,6 +376,12 @@
   #error "USB_SD_DISABLED is now NO_SD_HOST_DRIVE. Please update your Configuration_adv.h."
 #elif defined(USB_SD_ONBOARD)
   #error "USB_SD_ONBOARD is obsolete. Disable NO_SD_HOST_DRIVE instead."
+#elif POWER_SUPPLY == 1
+  #error "Replace POWER_SUPPLY 1 by enabling PSU_CONTROL and setting PSU_ACTIVE_HIGH to 'false'."
+#elif POWER_SUPPLY == 2
+  #error "Replace POWER_SUPPLY 2 by enabling PSU_CONTROL and setting PSU_ACTIVE_HIGH to 'true'."
+#elif defined(POWER_SUPPLY)
```
Comment (AnHardt) [inline Marlin/src/inc/SanityCheck.h:383]: Wow. Great. That gives us the chance to completely get rid of the comment in the configs (// Set 'false' for ATX (1), 'true' for X-Box (2)).

File: Marlin/Configuration.h [hunk 4]
```diff
@@ -312,22 +312,20 @@
...
- * :{ 0:'No power switch', 1:'ATX', 2:'X-Box 360' }
+ * Enable and connect the power supply to the PS_ON_PIN.
+ * Specify whether the power supply is active HIGH or active LOW.
  */
-#define POWER_SUPPLY 0
+//#define PSU_CONTROL
+//#define PSU_NAME "Power Supply"
+
+#if ENABLED(PSU_CONTROL)
+  #define PSU_ACTIVE_HIGH false // Set 'false' for ATX (1), 'true' for X-Box (2)
```
Comment (AnHardt) [inline Marlin/Configuration.h:324]: How about something more general here?
"// If M80 switches off the PSU change to the other logic state here." ?

Comment (thinkyhead) [reply Marlin/Configuration.h:324]: It's a comment for users migrating from older configs.

File: Marlin/src/lcd/menu/menu_info.cpp (hunk 2 above)
Comment (thinkyhead) [reply Marlin/src/lcd/menu/menu_info.cpp:179]: I just merge 'em, and somebody not us wanted this. But I agree, it is not pertinent to firmware.

File: Marlin/src/inc/SanityCheck.h (hunk 3 above)
Comment (thinkyhead) [reply Marlin/src/inc/SanityCheck.h:383]: Indeed. A more general comment will be nicer.
As usual, I will follow up on the things you suggest that I should do as time allows.

**Pull request merged**